[17.0][MIG] analytic_tag_dimension: Migration to 17.0

MaxyMoos opened the pull request.

angelmoya committed:
ADD Analytic Dimensions Module

angelmoya committed:
IMP analytic_dimension_id on account.analytic.tag not required

Commit:
[10.0] analytic_tag_dimension: fix/improve README (#139)

Commit:
[MIG] analytic_tag_dimension: Migrate to 11.0

* [REF] analytic_tag_dimension: Improve additional fields creation
* [UX] analytic_tag_dimension: Warn user about spaces in codes
* [FIX] analytic_tag_dimension: fix tests
* [IMP] analytic_tag_dimension: make tests SavepointCase

Commit:
[FIX] analytic_tag_dimension: grouping in report->invoice

Commit:
[MIG] analytic_tag_dimension: Migration to 12.0

carlosdauden committed:
[IMP] analytic_tag_dimension: Sync dimension code and description changes

eLBati committed:
FIX analytic_tag_dimension when inserting upper case code

Saran440 committed:
[IMP] analytic_tag_dimension: black, isort, prettier

Saran440 committed:
[13.0][MIG] analytic_tag_dimension

pedrobaeza committed:
[FIX] analytic_tag_dimension: avoid column not exists

Creating dimensions, you can face this error:

psycopg2.ProgrammingError: column ail.x_dimension_xxx does not exist
LINE 48:         , ail.x_dimension_xxx as x_dimension_xxx, ai.team_..

This is because the column is not yet populated in the moment when
the model initialization is triggered.

In this commit, we inhibit the population of that field in the report
on that case, and launching manual initialization later when done.

pedrobaeza committed:
[REF] analytic_tag_dimension: Follow OCA guidelines of one file per main model

pedrobaeza committed:
[REF] analytic_tag_dimension: Refactor tests

- More simplicity
- Don't use demo data
- Test all conditions on each model

pedrobaeza committed:
[IMP] analytic_tag_dimension: Don't write 2 times when using tags

Previous code was doing a deferred write after the creation/first write for easing the
code, but it's very bad for performance in such used objects.

This commits changes the way this is handled for covering the cases in advance, and thus
not requiring a second write. The only drawback is that writing on multiple records
means to perform a write on each record separately. This case was also happening
previously, and we have gained one write (the initial one), but it's not the ideal one
in terms of performance. A possible improvement can be to group the records with the
same final values to be performed at once.

pedrobaeza committed:
[FIX] analytic_tag_dimension: Handle removal tags

When you remove tags, dimensions fields were not updated accordingly before this commit.

pedrobaeza committed:
[FIX] analytic_tag_dimension: Handle dimension removal and module uninstallation

If not, there is remaining garbage in ir.model.fields and SQL columns in tables, avoiding
a second install of the module or to reuse the same dimension code.

pedrobaeza committed:
[IMP] analytic_tag_dimension: Promote Tecnativa to co-author

After all the previous works, it's fair to do it.

Saran440 committed:
[14.0][MIG] analytic_tag_dimension

Saran440 committed:
[FIX] update invoice_report direct databbase

cesar-tecnativa committed:
[MIG] analytic_tag_dimension: Migration to 15.0

Commit:
[UPD] Update analytic_tag_dimension.pot

OCA-git-bot committed:
[UPD] README.rst

Saran440 committed:
[FIX] test script savepointcase to transactioncase

Saran440 committed:
[FIX] account_tag_dimension: fix permission sudo

- Allow modify model 'ir.model'

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: account-analytic-15.0/account-analytic-15.0-analytic_tag_dimension
Translate-URL: https://translation.odoo-community.org/projects/account-analytic-15-0/account-analytic-15-0-analytic_tag_dimension/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
analytic_tag_dimension 15.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-analytic-15.0/account-analytic-15.0-analytic_tag_dimension
Translate-URL: https://translation.odoo-community.org/projects/account-analytic-15-0/account-analytic-15-0-analytic_tag_dimension/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: account-analytic-15.0/account-analytic-15.0-analytic_tag_dimension
Translate-URL: https://translation.odoo-community.org/projects/account-analytic-15-0/account-analytic-15-0-analytic_tag_dimension/it/

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (23 of 23 strings)

Translation: account-analytic-15.0/account-analytic-15.0-analytic_tag_dimension
Translate-URL: https://translation.odoo-community.org/projects/account-analytic-15-0/account-analytic-15-0-analytic_tag_dimension/pt_BR/

Commit:
Translated using Weblate (Catalan)

Currently translated at 100.0% (23 of 23 strings)

Translation: account-analytic-15.0/account-analytic-15.0-analytic_tag_dimension
Translate-URL: https://translation.odoo-community.org/projects/account-analytic-15-0/account-analytic-15-0-analytic_tag_dimension/ca/

hhgabelgaard committed:
Added translation using Weblate (Danish)

MaxyMoos committed:
[IMP] analytic_tag_dimension: pre-commit auto fixes

MaxyMoos committed:
[MIG] analytic_tag_dimension: Migration to 17.0

Saran440 commented:
@MaxyMoos Functional Tested. It's work :+1:  but display is not correct. can you check please?
![Selection_004](https://github.com/user-attachments/assets/60a4af4c-89b8-43a0-90f1-d535404bf76d)

Saran440 commented:
@MaxyMoos maybe v17 should fixed invoice report because `_sub_select` is not used. and test script is not cover

I updated in version18 already -> https://github.com/OCA/account-analytic/pull/761/commits/c3558d053bb6789aadcfc16568dea034829a71d5

MaxyMoos committed:
analytic_tag_dimension: Fix dimension form view + refactor invoice
report

MaxyMoos commented:
@Saran440 Hello, thanks a lot for checking out the PR :+1: 

Indeed the form view had issues, my bad, sorry about this.

I fixed it and added your refactor and test (I had to adapt the code, composing the SQL queries as you did in v18 did not work in v17 for some reason). If everything looks OK to you, I could then squash this new commit with the main migration one.

Saran440 reviewed (approved):
Functional Test 👍

dreispt reviewed (commented):
Aren't Analytic Dimensions an out of the box feature, supported through Analytic Plans?

MaxyMoos commented:
A similar functionality can certainly be achieved for many use cases with the Analytic Plans + Analytic Accounts, but since Analytic Tags are sometimes being used differently and a bit more freely than Accounts, "duplicating" this feature onto them makes sense - in a similar vein, the module `account_analytic_tag_distribution` (v16) has brought the Analytic Distribution over to Analytic Tags.

MaxyMoos committed:
analytic_tag_dimension: handle all m2m value formats

Some code paths trigger `create` or `write` calls that pass m2m values
as pure IDs (integers) rather than the classic `Command` tuple (x,y,z)

Andrii9090-tecnativa commented:
Hi, @Saran440 !
Can you create a PR with migration to 18 with you [commit](https://github.com/OCA/account-analytic/commit/c3558d053bb6789aadcfc16568dea034829a71d5) ?

Thanks!
TT60956

Saran440 commented:
@Andrii9090-tecnativa I fixed and migrated to v18 already.
However, the PR was marked as stale.
Could someone please help reopen it? https://github.com/OCA/account-analytic/pull/761
